### PR TITLE
#25591 & character in SKUs is shown as &amp; in current variations li…

### PR DESCRIPTION
**Pull request by KaushikChavda**
…st on configurable products

 - Fixed the issue SKU special character which change to encoded version.

### Fixed Issues (if relevant)

1. magento/magento2#25591: & character in SKUs is shown as &amp; in current variations list on configurable products

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/ConfigurableProduct/view/adminhtml/web/js/components/dynamic-rows-configurable.js [hunk 1]
```diff
@@ -219,7 +219,7 @@ define([
             _.each(tmpData, function (row, index) {
                 path = this.dataScope + '.' + this.index + '.' + (this.startIndex + index);
                 row.attributes = $('<i></i>').text(row.attributes).html();
-                row.sku = $('<i></i>').text(row.sku).html();
+                row.sku = row.sku;
```
Comment (swnsma) [inline app/code/Magento/ConfigurableProduct/view/adminhtml/web/js/components/dynamic-rows-configurable.js:222]:
```suggestion
                row.sku = $('<i></i>').html(row.sku).html();
```

File: app/code/Magento/ConfigurableProduct/view/adminhtml/web/js/components/dynamic-rows-configurable.js [hunk 2]
```diff
@@ -405,7 +405,7 @@ define([
                 'id': row.productId,
                 'product_link': row.productUrl,
                 'name': $('<i></i>').text(row.name).html(),
-                'sku': $('<i></i>').text(row.sku).html(),
```
Comment (swnsma) [inline app/code/Magento/ConfigurableProduct/view/adminhtml/web/js/components/dynamic-rows-configurable.js:408]:
```suggestion
                row.sku = $('<i></i>').html(row.sku).html();
```

**Review by swnsma — CHANGES_REQUESTED** [review]
Hello @KaushikChavda!
Thank you for your contribution!

May we keep '<i></i>' as it was before?
Please see my suggestions.

File: app/code/Magento/ConfigurableProduct/view/adminhtml/web/js/components/dynamic-rows-configurable.js (hunk 1 above)
Comment (KaushikChavda) [reply app/code/Magento/ConfigurableProduct/view/adminhtml/web/js/components/dynamic-rows-configurable.js:222]: Hello @swnsma,

If the SKU is like "C&A" with your solutions "$('<i></i>').html('C&A').html();" is become "Best &amp; Test" and that not the correct solution for it.

Thanks

File: app/code/Magento/ConfigurableProduct/view/adminhtml/web/js/components/dynamic-rows-configurable.js (hunk 2 above)
Comment (KaushikChavda) [reply app/code/Magento/ConfigurableProduct/view/adminhtml/web/js/components/dynamic-rows-configurable.js:408]: Hello @swnsma,

If the SKU is like "C&A" with your solutions "$('<i></i>').html('C&A').html();" is become "Best &amp; Test" and that not the correct solution for it.

Thanks

Comment (swnsma) [reply app/code/Magento/ConfigurableProduct/view/adminhtml/web/js/components/dynamic-rows-configurable.js:408]: In this point - yes.
I am not happy with `$('<i></i>')` anyway, so it is better option to check if issue which was fixed by adding `$('<i></i>')` will not be reproduced in this PR.

**Review by swnsma — APPROVED** [review]
QA could you please also verify that `MAGETWO-88649: Wrong swatches behavior` is not being reproduced with codebase of this PR?

**Comment by magento-engcom-team** [comment]
Hi @swnsma, thank you for the review.
[ENGCOM-6483](https://jira.corp.magento.com/browse/ENGCOM-6483) has been created to process this Pull Request
:eight_spoked_asterisk: @swnsma, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Charlie** [comment]
Hi @KaushikChavda 

Due to [Magento Definition of Done](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing_dod.html) all code must be covered by tests. Please cover your fix by automated tests.

Thanks!

**Comment by swnsma** [comment]
Hi @KaushikChavda!
Will you proceed with tests coverage?

**Review by swnsma — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @swnsma, thank you for the review.
[ENGCOM-6483](https://jira.corp.magento.com/browse/ENGCOM-6483) has been created to process this Pull Request

**Comment by swnsma** [comment]
@magento run all tests

**Comment by engcom-Golf** [comment]
Failed functional test not related to the changes in this PR, test fails on 2.4-develop branch.

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

The _amp_ text is not added to the sku
configurable_sku

**Pull request merged**